### PR TITLE
Make dart_coveralls strong-mode clean.

### PR DESCRIPTION
**Pull request by eseidel**
Added an analysis_options.yaml file, enabled strong_mode
and then burned through the issues from the analyzer.

### CODE REVIEW

**Comment by eseidel** [comment]
I have my editor currently set up to automatically apply dart-fmt on save.  If that's a problem I can revert those parts of the changes.

FYI @kevmoo (who I think has contributed here before).

**Comment by kevmoo** [comment]
@eseidel mixing format fixes w/ other things? In separate commits? 😱

File: analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+analyzer:
+  strong-mode: true
+
```
Comment (kevmoo) [inline analysis_options.yaml]: Nit: Remove indent whitespace

Comment (eseidel) [inline analysis_options.yaml]: Done.

**Comment by eseidel** [comment]
I don't have commit privs, plz squash and merge at your convenience. :D

Thanks!

**Pull request merged**